pkg/xray/resource_xray_license_policy_test.go

```diff
@@ -0,0 +1,361 @@
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "name", policyName),
+					resource.TestCheckResourceAttr(fqrn, "description", policyDesc),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.name", ruleName),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.build_failure_grace_period_in_days", strconv.Itoa(gracePeriod)),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.custom_severity", "High"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.fail_build", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.active", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.unscanned", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.allowed_licenses.0", "Apache-1.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.allowed_licenses.1", "Apache-2.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.multi_license_permissive", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLicensePolicy_createBannedLic(t *testing.T) {
+	policyName := "terraform-license-policy-4"
+	policyDesc := "policy created by xray acceptance tests"
+	ruleName := "test-license-rule-4"
+	gracePeriod := 5
+	resourceName := "policy-" + strconv.Itoa(randomInt())
+	fqrn := "xray_license_policy." + resourceName
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		CheckDestroy:      testAccCheckLicensePolicyDestroy(fqrn),
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccXrayLicensePolicy_createBanneddLic(resourceName, policyName, policyDesc, ruleName, gracePeriod),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "name", policyName),
+					resource.TestCheckResourceAttr(fqrn, "description", policyDesc),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.name", ruleName),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.build_failure_grace_period_in_days", strconv.Itoa(gracePeriod)),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.custom_severity", "High"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.fail_build", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.active", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.actions.0.block_download.0.unscanned", "true"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.banned_licenses.0", "Apache-1.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.banned_licenses.1", "Apache-2.0"),
+					resource.TestCheckResourceAttr(fqrn, "rules.0.criteria.0.multi_license_permissive", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLicensePolicy_createMultiLicensePermissiveFalse(t *testing.T) {
+	policyName := "terraform-license-policy-5"
```

Comment:
Why not use these in a map, and then `executeTemplate`?

Reply:
done